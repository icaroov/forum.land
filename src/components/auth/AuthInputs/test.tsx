import { render, screen } from '@testing-library/react'

import AuthInputs from '.'

describe('<AuthInputs />', () => {
  it('should render the heading', () => {
    const { container } = render(<AuthInputs />)

    expect(
      screen.getByRole('heading', { name: /AuthInputs/i })
    ).toBeInTheDocument()

    // expect(container.firstChild).toMatchSnapshot()
  })
})
