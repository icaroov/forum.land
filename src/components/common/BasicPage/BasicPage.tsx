import { Box } from '@chakra-ui/react'
import Head from 'next/head'
import { ReactNode } from 'react'

export type BasicPageProps = {
  children: ReactNode
  meta?: {
    title: string
  }
}

const BasicPage = ({ children, meta }: BasicPageProps) => {
  return (
    <>
      {meta && (
        <Head>
          <title>{`${meta.title} | Reddit`}</title>
        </Head>
      )}

      <Box>{children}</Box>
    </>
  )
}

export default BasicPage
