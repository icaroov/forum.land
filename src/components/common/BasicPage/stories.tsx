import { ComponentMeta, ComponentStory } from '@storybook/react'

import BasicPage from '.'

export default {
  title: 'BasicPage',
  component: BasicPage
} as ComponentMeta<typeof BasicPage>

export const Default: ComponentStory<typeof BasicPage> = () => <BasicPage />
