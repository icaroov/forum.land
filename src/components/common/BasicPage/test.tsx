import { render, screen } from '@testing-library/react'

import BasicPage from '.'

describe('<BasicPage />', () => {
  it('should render the heading', () => {
    const { container } = render(<BasicPage />)

    expect(
      screen.getByRole('heading', { name: /BasicPage/i })
    ).toBeInTheDocument()

    // expect(container.firstChild).toMatchSnapshot()
  })
})
