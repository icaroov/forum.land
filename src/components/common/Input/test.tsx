import { render, screen } from '@testing-library/react'

import Input from '.'

describe('<Input />', () => {
  it('should render the heading', () => {
    render(<Input />)

    expect(screen.getByRole('heading', { name: /Input/i })).toBeInTheDocument()

    // expect(container.firstChild).toMatchSnapshot()
  })
})
