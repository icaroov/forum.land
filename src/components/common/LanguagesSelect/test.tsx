import { render, screen } from '@testing-library/react'

import LanguagesSelect from '.'

describe('<LanguagesSelect />', () => {
  it('should render the heading', () => {
    render(<LanguagesSelect />)

    expect(
      screen.getByRole('heading', { name: /LanguagesSelect/i })
    ).toBeInTheDocument()

    // expect(container.firstChild).toMatchSnapshot()
  })
})
