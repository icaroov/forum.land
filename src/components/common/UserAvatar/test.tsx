import { render, screen } from '@testing-library/react'

import { UserType } from '@src/types/user'

import UserAvatar from '.'

const userMock: Partial<UserType> = {
  displayName: 'John Doe',
  email: '',
  emailVerified: true,
  isAnonymous: false,
  metadata: {
    creationTime: '',
    lastSignInTime: ''
  },
  phoneNumber: '',
  photoURL: '',
  providerData: [],
  providerId: '',
  refreshToken: '',
  tenantId: '',
  uid: ''
}

describe('<UserAvatar />', () => {
  it('should render the heading', () => {
    const { container } = render(<UserAvatar user={userMock} />)

    expect(
      screen.getByRole('heading', { name: /UserAvatar/i })
    ).toBeInTheDocument()

    // expect(container.firstChild).toMatchSnapshot()
  })
})
