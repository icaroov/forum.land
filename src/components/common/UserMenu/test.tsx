import { render, screen } from '@testing-library/react'

import UserMenu from '.'

describe('<UserMenu />', () => {
  it('should render the heading', () => {
    render(<UserMenu />)

    expect(
      screen.getByRole('heading', { name: /UserMenu/i })
    ).toBeInTheDocument()

    // expect(container.firstChild).toMatchSnapshot()
  })
})
