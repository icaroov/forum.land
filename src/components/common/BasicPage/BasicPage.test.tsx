import { screen } from '@testing-library/react'

import { renderWithProviders } from '@src/config/test/renderWithProviders'

import BasicPage from '.'

describe('<BasicPage />', () => {
  it('should render the component correctly', () => {
    const { container } = renderWithProviders(
      <BasicPage
        meta={{
          title: 'Test'
        }}
      >
        <h1>Test</h1>
      </BasicPage>
    )

    expect(container).toMatchSnapshot()
    expect(screen.getByText('Test')).toBeInTheDocument()
  })
})
