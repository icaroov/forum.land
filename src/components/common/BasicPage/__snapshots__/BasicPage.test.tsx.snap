// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BasicPage /> should render the component correctly 1`] = `
<div>
  <div
    class="css-0"
  >
    <h1>
      Test
    </h1>
  </div>
</div>
`;
